src/server/editorServices.ts

```diff
@@ -1973,7 +1975,9 @@ namespace ts.server {
         /* @internal */
         private createLoadAndUpdateConfiguredProject(configFileName: NormalizedPath, reason: string) {
             const project = this.createAndLoadConfiguredProject(configFileName, reason);
-            project.updateGraph();
+            project.updateGraph(/*skipEvents*/ true); // Defer telemetry and loading event until after AutoImportProvider is created
+            project.getPackageJsonAutoImportProvider(); // Preload AutoImportProviderProject to avoid creating it on first completions request
+            project.onFinishedLoading(); // Fire skipped events

```

Comment:
I dont think this is right way to do it.. You want to  handle it in updateGraph of main project instead.
Here are scenarios when project is created but not loaded right away but loaded when file is opened for that project

- Opening external project with lazyConfiguredProjectsFromExternalProject preference set to true
- config project structure like
serivices/tsconfig
services/compiler/tsconfig
Then you open a file from compiler and then open file from services